readme.md

```diff
@@ -70,6 +70,25 @@ pify(request, {multiArgs: true})('https://sindresorhus.com').then(result => {
 });
 ```
 
+##### allArgs
+
+Type: `boolean`
+Default: `false`
+
+By default, the promisified function will only return the second argument from the callback. Turning this on will make it return an array of all arguments from the callback, *including* the error argument. This can be useful if you want to avoid a try-catch block when using async-await.
```

Comment:
Thanks! My usecase is not related to request (I usually use `wreck` or `fetch` but it's the first time I hear about `got`. Will definitely check it out).

My usecase is more about a series of awaits and how it looks when you have to wrap them in try- catch. I know that always just resolving is not naturally 'promisy' but just feels better in some cases (less nesting):

```js
try {
  await pify(step1())
  const [err] = await pify(step2(), {allArgs: true})
  if(err) { 
     // handle specific error
  }
} catch (e) {
  //handle other errors
}
```
vs

```js
try {
  await pify(step1())
  try {
    await pify(step2())
  } catch (e) {
    //handle other errors
  }
} catch (e) {
  //handle other errors
}
```

A nice side-effect of the `allArgs` flag is that non-error first APIs (yeah, I know, how likely is that) can also be supported.